enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -387,23 +542,69 @@ None
 
 ### Implementation Details/Notes/Constraints
 
+#### New API changes for GA release
+
+Below new fields are added in `istiocsrs.operator.openshift.io` to support functionality not included in TP and also
+based on user feedback.
+- `spec.istioCSRConfig.istioDataPlaneNamespaceSelector` - `istio-csr` agent by default creates a configmap with the
+  CA certificate used for signing the istio server certificate and is required by the proxies in the data plane to verify
+  the server certificate to enable mTLS. `istio-csr` agent just creates and updates the configmap `istio-ca-root-cert`
+  and does not delete/removes it. Creating the configmap in all namespaces would not be required for and namespaces created
+  for OpenShift control plane like `kube-system`, and those starting `openshift` can be excluded. `istioDataPlaneNamespaceSelector`
+  helps in filtering the namespaces. Please refer `Risks and Mitigations` section for other details.
+- `spec.istioCSRConfig.certManager.istioCACertificate` - In the TP release, operator was extracting the CA certificate
+  from the configured self-signed issuer and was making it available in the container using volumes. But for issuer types
+  like vault, venafi, the CA certificate used is not available in issuer. `istioCACertificate` allows configuring the configmap
+  which contains the CA certificate used for signing the istiod server certificates, and it will be made available to `istio-csr`
+  using volumes, after basic validations like certificate is indeed CA and the CA attribute in certificates `Basic Constraints`
+  section it set to true and was used for signing the istiod server certificate. When `istioCACertificate` is not provided
```

Comment:
Should we also mention that, despite whatever configMap key is configured, the controller will always mount the certificate at `/var/run/configmaps/istio-csr/ca.crt` path and set `--root-ca-file` arg in the istio-csr deployment accordingly. Additionally, the VolumeName would be `root-ca`

Reply:
I think it's not required, because user is not dependent on these or need to be aware of it.